### PR TITLE
MAINT: Change misleading names related to restarting simulations

### PR DESCRIPTION
**Pull request by IngridKJ**
## Proposed changes

Changes proposed by @pschultzendorff and I. Addresses issue #1199. 

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: src/porepy/numerics/time_step_control.py [hunk 1]
```diff
@@ -400,16 +400,16 @@ def __init__(
         self._iters: Union[int, None] = None
 
         # Book keeping of saved time steps for restarting purposes.
```
Comment (IvarStefansson) [inline src/porepy/numerics/time_step_control.py:402]:
```suggestion
        # Bookkeeping of saved time steps for restarting purposes.
```

File: src/porepy/numerics/time_step_control.py [hunk 2]
```diff
@@ -736,10 +736,10 @@ def write_time_information(self, path: Optional[Path] = None) -> None:
         """
 
         # Book keeping
```
Comment (IvarStefansson) [inline src/porepy/numerics/time_step_control.py:738]:
```suggestion
        # Bookkeeping
```

**Review by IvarStefansson — COMMENTED** [review]
Looks good to me

**Review by jwboth — APPROVED** [review]
I agree with @IvarStefansson  and have nothing to add. @Yuriyzabegaev misliked the documentation (feel free to shout and be more specific), yet to me the changes look consistent and sufficient. Thanks!

**Comment by Yuriyzabegaev** [comment]
The changes look absolutely reasonable for me, thanks for your efforts!

**Pull request merged**